src/Features/Core/Portable/PullMemberUp/QuickAction/AbstractMemberPullerWithQuickAction.cs

```diff
@@ -1,187 +0,0 @@
-            foreach (var analysisResult in result.MemberAnalysisResults)
-            {
-                foreach (var syntax in syntaxMap[analysisResult.Member])
-                {
-                    var originalMemberEditor = editorMap[syntax.SyntaxTree];
-                    originalMemberEditor.RemoveNode(originalMemberEditor.Generator.GetDeclaration(syntax));
-                }
-            }
-
-            return solutionEditor.GetChangedSolution();
-        }
-
-        private async Task<(EditorMap editorMap, SyntaxMap syntaxMap)> InitializeEditorMapsAndSyntaxMapAsync(
-            PullMembersUpAnalysisResult result,
-            SyntaxNode destinationSyntaxNode,
-            SolutionEditor solutionEditor,
-            Solution solution,
-            CancellationToken cancellationToken)
-        {
-            // Members and destination may come from different documents,
-            // EditorMap is used to save and group all the editors will be used.
-            var editorMapBuilder = ImmutableDictionary.CreateBuilder<SyntaxTree, DocumentEditor>();
-            // One member may have multiple syntaxNodes (e.g partial method).
-            // SyntaxMap is used to find the syntaxNodes need to be changed more easily.
-            var syntaxMapBuilder = ImmutableDictionary.CreateBuilder<ISymbol, SyntaxNode[]>();
-
-            foreach (var memberAnalysisResult in result.MemberAnalysisResults)
-            {
-                var tasks = memberAnalysisResult.Member.DeclaringSyntaxReferences.SelectAsArray(@ref => @ref.GetSyntaxAsync(cancellationToken));
-                var allSyntaxes = await Task.WhenAll(tasks).ConfigureAwait(false);
-                await AddEditorsToEditorMapBuilderAsync(editorMapBuilder, allSyntaxes, solutionEditor, solution).ConfigureAwait(false);
-                syntaxMapBuilder.Add(memberAnalysisResult.Member, allSyntaxes);
-            }
-
-            await AddEditorsToEditorMapBuilderAsync(
-                editorMapBuilder,
-                new[] { destinationSyntaxNode },
-                solutionEditor,
-                solution).ConfigureAwait(false);
-            return (editorMapBuilder.ToImmutableDictionary(), syntaxMapBuilder.ToImmutableDictionary());
-        }
-
-        private async Task AddEditorsToEditorMapBuilderAsync(
-            ImmutableDictionary<SyntaxTree, DocumentEditor>.Builder mapBuilder,
-            SyntaxNode[] syntaxNodes,
-            SolutionEditor solutionEditor,
-            Solution solution)
-        {
-            foreach (var syntax in syntaxNodes)
-            {
-                if (!mapBuilder.ContainsKey(syntax.SyntaxTree))
-                {
-                    var editor = await solutionEditor.GetDocumentEditorAsync(solution.GetDocumentId(syntax.SyntaxTree)).ConfigureAwait(false);
-                    mapBuilder.Add(syntax.SyntaxTree, editor);
-                }
-            }
-        }
-    }
-}
```

Comment:
move this code made this PR much harder to read.  Is it possible to do this in a way that doesn't make the PR problematic?

Reply:
for example, use partial types, and keep the code in this file.